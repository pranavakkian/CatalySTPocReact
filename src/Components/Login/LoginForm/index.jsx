import PropTypes from 'prop-types';
import { useState } from 'react';
import { Button, Form } from 'semantic-ui-react';

const LoginForm = (props) => {
  const { handleLoginForm } = props;
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  return (
    <div>
      <Form>
        <Form.Field>
          <input id="username" placeholder="Username" label="Username" value={username} onChange={(e) => setUsername(e.target.value)} />
        </Form.Field>
        <Form.Field>
          <input placeholder="Password" value={password} label="Password" onChange={(e) => setPassword(e.target.value)} />
        </Form.Field>
        <Button type="submit" onClick={() => handleLoginForm(username, password)}>Submit</Button>
      </Form>
    </div>
  );
};

LoginForm.propTypes = {
  handleLoginForm: PropTypes.func.isRequired,
};
export default LoginForm;
