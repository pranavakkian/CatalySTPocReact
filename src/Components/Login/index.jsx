import React from 'react';

const loginPage = () => (
  <h1>Login Page.</h1>
);

export default loginPage;
