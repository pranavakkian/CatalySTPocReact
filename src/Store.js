import { applyMiddleware, createStore } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import RootReducer from './RootReducer';

const composeEnhancers = composeWithDevTools({ trace: true });

const Store = createStore(RootReducer, composeEnhancers(applyMiddleware()));

export default Store;
