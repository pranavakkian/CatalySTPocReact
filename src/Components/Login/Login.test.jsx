import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';
import configureMockStore from 'redux-mock-store';
import Login from './index';
import Store from '../../Store';

describe('Login', () => {
  const MockStore = configureMockStore();
  const mockstore = MockStore(() => ({
    UserInfoReducer: {
      isLoggedIn: true,
      userName: 'test',
    },
  }));
  const mockstore2 = MockStore(() => ({
    UserInfoReducer: {
      isLoggedIn: false,
      userName: null,
    },
  }));
  test('Component rendered successfully', () => {
    const { container } = render(<Provider store={mockstore}><Login /></Provider>);
    expect(container).toMatchSnapshot();
  });
  test('Login Failed', () => {
    // userEvent.type(screen.getAllByRole('textbox')[0], '');
    // userEvent.type(screen.getAllByRole('textbox')[1], '');
    render(<Provider store={mockstore2}><Login /></Provider>);
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(screen.getByText('Invalid Credentials.')).toBeInTheDocument();
    userEvent.click(screen.getByRole('button', { name: 'OK' }));
    const { UserInfoReducer } = mockstore2.getState();
    expect(UserInfoReducer).toStrictEqual({ isLoggedIn: false, userName: null });
  });
  test('Login Successful', () => {
    render(<Provider store={mockstore}><Login /></Provider>);
    userEvent.type(screen.getAllByRole('textbox')[0], 'test');
    userEvent.type(screen.getAllByRole('textbox')[1], 'admin123');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    const { UserInfoReducer } = mockstore.getState();
    expect(UserInfoReducer).toStrictEqual({ isLoggedIn: true, userName: 'test' });
  });
  test('Login Successful', () => {
    render(<Provider store={mockstore}><Login /></Provider>);
    userEvent.type(screen.getAllByRole('textbox')[0], 'abc');
    userEvent.type(screen.getAllByRole('textbox')[1], 'abc123');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    const { UserInfoReducer } = mockstore.getState();
    expect(UserInfoReducer).toStrictEqual({ isLoggedIn: true, userName: 'test' });
  });
});
