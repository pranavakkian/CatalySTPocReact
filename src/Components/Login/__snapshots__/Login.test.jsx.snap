// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component rendered successfully 1`] = `
<div>
  <div>
    <h1>
      Login Page.
    </h1>
    <div>
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <input
            id="username"
            label="Username"
            placeholder="Username"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <input
            label="Password"
            placeholder="Password"
            value=""
          />
        </div>
        <button
          class="ui button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <button
      class="ui button"
    >
      Show Modal
    </button>
  </div>
</div>
`;
